controllers/prometheusrule_controller.go

```diff
@@ -146,24 +146,26 @@ func (r *PrometheusRuleReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 				if rule.Alert != "" {
 					a = strings.ToLower(rule.Alert)
 					if _, ok := alertMap[a]; !ok {
-						alertMap[a] = []prometheus.Rule{rule} 
+						alertMap[a] = []prometheus.Rule{rule}
 						continue
 					}
 					alertMap[a] = append(alertMap[a], rule)
-				}	
+				}
 			}
 		}
 
+		alertsNotToRemove := make(map[string]bool)
```

Comment:
```suggestion
		alertsToKeep := make(map[string]bool)
```

We could improve readability by removing the negation from the variable name.

This can get confusing as we use `!alertsNotToRemove` further below, reading as `not not to remove`